### ISSUE TITLE
NumberValidator not using config defined decimal separator

### ISSUE THREAD

**Issue by lluisclava** [yiisoft/yii2#18312]
NumberValidator is not using the config defined decimal separator

### What steps will reproduce the problem?
Server with english locale and no PHP intl extension installed.
Set in common/main.php
    'language'=>'ca-ES',
    'sourceLanguage'=>'ca-ES',
    'timeZone'=>'Europe/Andorra',//Posar andorra i si va malament Madrid.
    'charset'=>'UTF-8',
    'components' => [
        'formatter' => [
            'locale'=>'ca-ES',
            'dateFormat' => 'dd/MM/yyyy',
            'datetimeFormat' => 'php:d/m/Y - H:i',
            'currencyCode' => 'EUR',
            'thousandSeparator' => '.',
            'decimalSeparator' => ','
        ],

### What is the expected result?
When you use the numbervalidator in a form, the number format validated is with decimal separator "."

### What do you get instead?
When you use the numbervalidator in a form, the number format validated is with decimal separator ","

### Additional info
I think the problem resides in the function StringHelper::normalizeNumber that is used in NumberValidator. In this function it uses localeconv() php function to get the locale. Yii2 never uses the setlocale function to set the locale with the defined configuration in config file.

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.35
| PHP version      | 7.2.14
| Operating system | Linux

**Comment by bizley** [comment]
But NumberValidator is not using formatter at all.

**Comment by lluisclava** [comment]
Yes but, Yii2 is not getting the correct locale from configuration file and is an inconsistent behaivour. Because the application's output data is formatted with locale while the input data not.

**Comment by bizley** [comment]
So you are suggesting that NumberValidator should use the formatter, yes? Could you propose the solution PR?

**Comment by lluisclava** [comment]
Yes. My solutiion is NumberValidator should use the locale configuration defined in config file. With this, the application will work consistently (about input and output data format).

I would like to porpouse the code to solve this, but I don't know what is the correct way to get the locale configuration from yii2 application objects.

The thing is NumberValidator in line 90 i 110 it's using  StringHelper::normalizeNumber($value). This method uses $localeInfo = localeconv(); to get the locale configuration. I thing this should be changed for using the configs locale defined.

Thanks

**Comment by bizley** [comment]
@yiisoft/core-developers thoughts?

**Comment by samdark** [comment]
I've revised normalization for Yii 3 and it seems we actually can do it without locale: https://github.com/yiisoft/strings/blob/master/src/NumericHelper.php#L48 We can backport such approach...
